src/main/java/org/opensearch/security/OpenSearchSecurityPlugin.java

```diff
@@ -339,6 +341,17 @@ public Object run() {
             deprecationLogger.deprecate("Setting {} is ignored.", advancedModulesEnabledKey);
         }
 
+        final boolean unsupportedRestapiAllowSecurityConfigModification = settings.getAsBoolean(
+            SECURITY_UNSUPPORTED_RESTAPI_ALLOW_SECURITYCONFIG_MODIFICATION,
+            false
+        );
+        if (unsupportedRestapiAllowSecurityConfigModification) {
+            deprecationLogger.deprecate(
+                "Setting {} is deprecated. Please use permissions instead.",
+                SECURITY_UNSUPPORTED_RESTAPI_ALLOW_SECURITYCONFIG_MODIFICATION
+            );
+        }
+
```

Comment:
See `DeprecatedSettings.checkForDeprecatedSetting(...)` in this codebase

Reply:
Fixed